src/icon-server/fastify-plugin.js

```diff
@@ -1,25 +1,28 @@
 import { parseVersionId } from '@mapeo/schema'
 import { Type as T } from '@sinclair/typebox'
 import fp from 'fastify-plugin'
+import { kDataTypes, kCoreManager } from '../mapeo-project.js'
 
 export default fp(iconServerPlugin, {
   fastify: '4.x',
   name: 'mapeo-icon-server',
 })
 
-const HEX_REGEX_32_BYTES = '^[0-9a-fA-F]{64}$'
-const HEX_STRING_32_BYTES = T.String({ pattern: HEX_REGEX_32_BYTES })
+// const HEX_REGEX_32_BYTES = '^[0-9a-fA-F]{64}$'
+// const HEX_REGEX_26_BYTES = '^[0-9a-fA-F]{52}$'
+const HEX_STRING_32_BYTES = T.String({})
```

Comment:
I commented this regexes 'cause I was getting validation errors and I couldn't bother with regex today.
It seems that the project key is actually 26 bytes? (printing `projectId` that's returned from `manager.createProject()` yields `52`...)
Anyways, there's probably something I'm missing...

Reply:
that number seems familiar for some reason, but the project id that's used for the url _should_ be 32 bytes since we're using this from @mapeo/crypto to create them:

https://github.com/digidem/mapeo-crypto/blob/a3ec70b97318260e2f1593d6abbaeda065cc9407/utils.js#L39

Reply:
ah yeah the encoding is different, which produces an id of 26 bytes. The projectId here is really the project's "public" id, which is a one-way hashed version of the project's "internal" id. Think it's a matter of updating this regex to assume length of 52 characters

Needs to be done for the blob server plugin too (but that's out of scope here)

Reply:
Mhm, following the trail, the `keys.projectKey` on `mapeo-manager:217` seems to be 32-bytes long. Now, after passing it to `projectKeyToPublicId` (which wraps `keyToPublicId`) the resulting string length is 52 (shouldn't it be 64?).
I see in that fn that you linked that it uses some encoding called z-base-64. I'm not familiar with that, but does that changes something??
I see that I'm on mapeo-crypto-alpha.8 which is the last version, but at the same time on my local version of `mapeo-crypto` that function is named `projectKeyToPublicId` instead of `keyToPublicId` (I've `rm -rf node_modules && npm install` just in case...)

Reply:
@tomasciccola yeah not sure if you saw my last comment but the project id we're returning from `createProject` is not a hex-encoded string but rather the z-base-64 one. this is expected, i just forgot that it's a different length and the only change needed here is updating the regex to use 52 instead of 64, since this API would only be accepting "public" ids

Reply:
also updating the variable name to something like `PROJECT_PUBLIC_ID_REGEX`, maybe with the addition of a comment to summarize what's been explained here 😄

Reply:
yeah, an also include every letter since its not hex (I think only lowercase, but left also upper just to be safe)